lib/base/application.cpp

```diff
@@ -180,7 +180,7 @@ void Application::SetResourceLimits(void)
 	rlimit rl;
 
 #	ifdef RLIMIT_NOFILE
-	rl.rlim_cur = 16 * 1024;
+	rl.rlim_cur = GetRLimitFiles();
```

Comment:
I would set the current values as minimal. Everything lower would fail, only setting to `0` or `NULL` should be valid.

Then it's the users problem.